js/filter.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+(function () {
+  var OFFERS_DOWNLOAD_URL = 'https://js.dump.academy/keksobooking/data';
+  var MAX_PINS_ON_MAP = 5;
+  var offersMap = document.querySelector('.map');
+  var mapPins = offersMap.querySelector('.map__pins');
+  var offersFilter = document.querySelector('.map__filters');
+  var housingTypeSelect = offersFilter.querySelector('#housing-type');
+  var offersData = [];
+
+  var refreshOffers = window.debounce(function () {
+    window.pins.clear();
+    window.pins.render(mapPins, filterOffers());
+  });
+
+  var filterOffers = function () {
+    var filteredOffers = [];
+    var offerType = housingTypeSelect.value;
+
+    filteredOffers = offersData.filter(function (item) {
+      if (offerType === 'any') {
+        return true;
+      } else {
+        return item.offer.type === offerType;
+      }
```

Comment:
Тут else нужно опустить, так как в ветке if есть return
```suggestion
      if (offerType === 'any') {
        return true;
      } 
      return item.offer.type === offerType;
```